app/models/partner.rb

```diff
@@ -202,6 +205,21 @@ def check_ward_access
     end
   end
 
+  def check_service_area_access
+    return unless service_areas.any?
+
+    user = User.where(id: accessed_by_id).first
+    return if user.blank?
+
+    my_neighbourhoods = service_area_neighbourhoods.pluck(&:id)
```

Comment:
pluck will return only the first service area neighbourhood, I think

Reply:
At least I had that exact problem when I used this in `owned_neighbourhood_ids`. You probably want `collect` or `map`, and I don't see an access to subtree so you're not accounting for neighbourhoods that are of a lower region service area (Might not be necessary?)

Reply:
- Coming back to this: I was wrong! :)  I dunno what weirdness happened with pluck last time I tried it to make it return only 1 item when map/collect returned all 600-odd, but it seems to work fine now. Chalking this up to eldrich horrors / rails trauma.

I think there might? be a difference between `&:id` and `:id` here, and you will want the latter. Just checked in console and the former gives some wacky output by comparison